contracts/protocol/clients/voucher/BosonVoucher.sol

```diff
@@ -217,42 +219,53 @@ contract BosonVoucherBase is IBosonVoucher, BeaconClientBase, OwnableUpgradeable
-        Range storage range = _rangeByOfferId[_offerId];
-
-        // Revert if id not associated with a range
-        require(range.length != 0, NO_RESERVED_RANGE_FOR_OFFER);
-
-        // Revert if no more to mint in range
-        require(range.length >= range.minted + _amount, INVALID_AMOUNT_TO_MINT);
-
-        // Get max amount that can be minted in a single transaction
-        address protocolDiamond = IClientExternalAddresses(BeaconClientLib._beacon()).getProtocolAddress();
-        uint256 maxPremintedVouchers = IBosonConfigHandler(protocolDiamond).getMaxPremintedVouchers();
-
-        // Revert if too many to mint in a single transaction
-        require(_amount <= maxPremintedVouchers, TOO_MANY_TO_MINT);
-
-        // Make sure that offer is not expired or voided
-        (Offer memory offer, OfferDates memory offerDates) = getBosonOffer(_offerId);
-        require(!offer.voided && (offerDates.validUntil > block.timestamp), OFFER_EXPIRED_OR_VOIDED);
-
-        // Get the first token to mint
-        uint256 start = range.start + range.minted;
-
-        // Pre-mint the range to the seller
-        uint256 tokenId;
         address seller = owner();
-        for (uint256 i = 0; i < _amount; i++) {
-            tokenId = start + i;
-            emit Transfer(address(0), seller, tokenId);
-        }
 
-        // Bump the minted count
-        range.minted += _amount;
+        preMintInternal(seller, _offerId, _amount);
+    }
 
-        // Update seller's total balance
-        getERC721UpgradeableStorage()._balances[seller] += _amount;
+    /**
+     * @notice Pre-mints all or part of an offer's reserved vouchers.
+     *
+     * For small offer quantities, this method may only need to be
+     * called once.
+     *
+     * But, if the range is large, e.g., 10k vouchers, block gas limit
+     * could cause the transaction to fail. Thus, in order to support
+     * a batched approach to pre-minting an offer's vouchers,
+     * this method can be called multiple times, until the whole
+     * range is minted.
+     *
+     * A benefit to the batched approach is that the entire reserved
+     * range for an offer need not be pre-minted at one time. A seller
+     * could just mint batches periodically, controlling the amount
+     * that are available on the market at any given time, e.g.,
+     * creating a pre-minted offer with a validity period of one year,
+     * causing the token range to be reserved, but only pre-minting
+     * a certain amount monthly.
+     *
+     * Caller must be contract owner (seller assistant address).
+     *
+     * Reverts if:
+     * - Offer id is not associated with a range
+     * - Amount to mint is more than remaining un-minted in range
+     * - Too many to mint in a single transaction, given current block gas limit
+     * - Offer already expired
+     * - Offer is voided
+     * - _to address is zero address
+     *
+     * @param _to - the address to mint the vouchers to
```

Comment:
If `_to` can be any address, this gives `assistant` probably too great power - they could mint vouchers to anybody (including themselves), and whoever the first owner is, gets the paid price (directly from marketplace), while in protocol it is sellers account that needs to provide the funds.

Also, as long as `preMint` that transfers funds to assistant exists, the problem of who gets the funds isn't solved.

I believe we don't even need two methods (`preMint` and `preMintTo`). Just `preMint` should be enough, where destination address should be some trusted address (for example `treasury`, `address(this)` or address that can be set only by the admin).

Reply:
As we discussed today, we could have two methods:
1. `preMint` (which could maybe be renamed to `premintToOwner`) and
2. `preMintToContract` which just mints vouchers to `address(this)`.